### ISSUE TITLE
raw_exec allows incorrect paramters in the config block

### ISSUE THREAD

**Issue by shantanugadgil** [hashicorp/nomad#5729]
### Nomad version
Nomad v0.9.1 (4b2bdbd9ab68a27b10c2ee781cceaaf62e114399)

### Operating system and Environment details
CentOS 7 x86_64 + updates

### Issue
A `raw_exec` job complains about the `port_map` **_after_** accepting the job to be `run`.
Job gets submitted but the allocation fails with the messed up error in output of `nomad alloc status NNNNN`

Removing/commenting the `port_map` makes the job work fine.

```
 failed to parse config
* :1,31-41: Extraneous JSON object property; No argument or block type is named "port_map".
```

### Reproduction steps
Use the following job file and run the job. The allocation will fail with the error.
Seem somewhat related to: https://github.com/hashicorp/nomad/issues/5308

### Job file (if appropriate)

```hcl
job "gowebhello" {
  datacenters = ["dc1"]

  group "group1" {

    task "webserver" {
      driver = "raw_exec"

    artifact {
         #source = "https://github.com/udhos/gowebhello/releases/download/v0.6/gowebhello_linux_amd64"
         source = "http://192.168.0.51:8080/www/gowebhello_linux_amd64"
         mode = "file"
         destination = "local/gowebhello"
      }

      config {
        command = "local/gowebhello"
        port_map {
          http = 8080
        }
      }

      resources {
        cpu    = 500
        memory = 256
        network {
          mbits = 10
          port "http" {}
        }
      }

      service {
        name = "gowebhello"
        tags = ["global", "cache"]
        port = "http"
        check {
          name     = "alive"
          type     = "tcp"
          interval = "10s"
          timeout  = "2s"
        }
      }
    }
  }
}
```

### Expected behaviour
The `nomad plan` and/or `nomad run` commands should flag the error and not let the job be submitted.

**Comment by cgbaker** [comment]
thanks, @shantanugadgil .

**Comment by shantanugadgil** [comment]
@cgbaker adding a message to keep the issue alive.

**Comment by cgbaker** [comment]
thanks, @shantanugadgil . we're still aware of this.

**Comment by bsftshantanu** [comment]
I think this can be closed as of Nomad version 1.2.0? :thinking:

**Comment by shantanugadgil** [comment]
This issue exists with version 1.9.1 as well. Seems not so important. It can be figured out using `nomad alloc status ...` to see why the allocation is stuck in `pending`

fwiw:

```
Recent Events:
Time                       Type                   Description
2024-11-06T19:46:25+05:30  Not Restarting         Error was unrecoverable
2024-11-06T19:46:25+05:30  Failed Validation      2 errors occurred:
	* failed to parse config: 
	* Invalid label: No argument or block type is named "port_map".

...
```
